Samples.Objc/Samples.ObjC/ViewControllers/AddItemViewController/AddItemViewController.m

```diff
@@ -0,0 +1,242 @@
+  [[self.titleLabel.bottomAnchor constraintEqualToAnchor:self.tableView.topAnchor constant:-16] setActive:true];
+  
+  [[self.tableView.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor] setActive:true];
+  [[self.tableView.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor] setActive:true];
+  [[self.tableView.bottomAnchor constraintEqualToAnchor:self.quantityView.topAnchor constant:-16] setActive:true];
+  
+  [[self.quantityView.leadingAnchor constraintEqualToAnchor:self.containerView.leadingAnchor constant:16] setActive:true];
+  [[self.quantityView.heightAnchor constraintEqualToConstant:40] setActive:true];
+  [[self.quantityView.widthAnchor constraintEqualToConstant:120] setActive:true];
+  [[self.quantityView.trailingAnchor constraintEqualToAnchor:self.saveButton.leadingAnchor constant:-16] setActive:true];
+  [[self.quantityView.centerYAnchor constraintEqualToAnchor:self.saveButton.centerYAnchor] setActive:true];
+  
+  self.saveButtonBottom = [self.saveButton.bottomAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.bottomAnchor constant:-16];
+  [[self.saveButton.trailingAnchor constraintEqualToAnchor:self.containerView.trailingAnchor constant:-16] setActive:true];
+  [self.saveButtonBottom setActive:true];
+  [[self.saveButton.heightAnchor constraintEqualToConstant:40] setActive:true];
+}
+
+// MARK: - UITableViewDelegate and UITableViewDataSource
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+  switch (indexPath.row) {
+    case 0:
+    {
+      TextFieldCell *cell = [[TextFieldCell alloc] initWithTitle:@"Item name" andPlaceholder:@"Enter item name" andText:self.itemName andKeyboardType:UIKeyboardTypeDefault];
+      cell.delegate = self;
+      return cell;
+    }
+    case 1:
+    {
+      TextFieldCell *cell = [[TextFieldCell alloc] initWithTitle:@"Price" andPlaceholder:@"Enter price" andText:self.itemPrice andKeyboardType:UIKeyboardTypeDecimalPad];
+      cell.delegate = self;
+      return cell;
+    }
+    case 2:
+    {
+      TextFieldCell *cell = [[TextFieldCell alloc] initWithTitle:@"Tax" andPlaceholder:@"Enter tax" andText:self.itemTax andKeyboardType:UIKeyboardTypeDecimalPad];
+      cell.delegate = self;
+      return cell;
+    }
+    default:
+    {
+      UITableViewCell *cell = [[UITableViewCell alloc] init];
+      return cell;
+    }
+  }
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
+  return UITableViewAutomaticDimension;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+  return 3;
+}
+
+// MARK: - TextFieldCellDelegate
+- (void)cell:(UITableViewCell *)cell textDidChange:(NSString *)text {
+  NSInteger row = [self.tableView indexPathForCell:cell].row;
+  switch (row) {
+    case 0:
+      self.itemName = text;
+      break;
```

Comment:
can remove

Reply:
I think maybe we should have break in each case, since it looks like for ObjC it will fall thru the next cases unless we have break.

Reply:
ah i had forgotten! good call